jina/peapods/pods/__init__.py

```diff
@@ -333,15 +255,14 @@ class Pod(BasePod):
     """
 
     class _ReplicaSet:
-        def __init__(self, pod_args: Namespace, args: List[Namespace]):
+        def __init__(
+            self, pod_args: Namespace, args: List[Namespace], head_args: Namespace
+        ):
             self.pod_args = copy.copy(pod_args)
+            self.head_args = head_args
```

Comment:
if a `ReplicaSet` needs to represent a set of Peas that can be also Heads, then I think `args` and `head_args` should be the same.

Reply:
or they should be Optional

Reply:
`ReplicaSet` can not  have heads, the head args are passed here because we need the address for activating/deactivating